### ISSUE TITLE
Clock intervalVariability

### ISSUE THREAD

**Issue by KarlWernersson** [modelica/fmi-standard#1663]
1. 
In section 2.4.4 table 15.
The attribute interval is wrong, it is called intervalVariablitliy

2. 
Are we missing an intevalVariability?
we have  
fixed, ->inteval set by fmu during initializtaion
calculated -> intevar computed by fmu from fixed parameters
tunable -> inteval computed by fmu from tunable parameters.  

Don't we miss a set tunable i.e interval->changeable from the FMU at event mode?
I think this was one of the scenarios in @rfranke original proposal.

Is this intentional or a mistake?

I propose that we

Change name of **tunable** to **calculatedTunable**, and **calculated** to **calculatedFixed**

Add a "new" variability **tunable** That allows setInteval in event mode and clock activation mode

An alternative would be to introduce an intevalCausality but i try to avoid that.

**Comment by KarlWernersson** [comment]
No, changing changes unregularly when the FMU want's it, what Ii am looking for is a clock where the period can change by calling setInterval from the importer during event mode. It is definitely not countdown, and triggered has nothing do do with interval so it is no longer an periodic clock.

To 1 Great =)

**Comment by KarlWernersson** [comment]
Just by calling an interval **tunable** I would assume that you can set it as you can do that with fixed, just as for fixed and tunable parameters, I was a bit surprised it did not work as I assumed and in opposite to fixed.

**Comment by KarlWernersson** [comment]
Working on the PR i realized that we only have **getShift** no **setShift** function and it is is only allowed for the **constant** and the **computed periodic clocks**, not the clocks where the interval is settable. 

Later on it says for attribute **shiftDecimal** and **shiftCounter** it states
This attribute must only be used for **periodic Clocks**
This is true but currently we are more restrictive and say that it must only be used for periodic clock with **clockVariability**
**constant**, **fixedComputed** (previously calculated) or **tunableCalculated**, (previously tunable)

**So my question is this**

Should we only support fraction for periodic clocks where the interval is set by the FMU, then it is  just a small upade for **shiftDecimal** and **shiftCounter** 

If not is it allowed for all periodic clocks?
Then I will update **fixed** and the new **tunable** accordingly.

Should it then also be possible to set the shift for **fixed** and **tunable**
If so we need the corresponding set functions.

**Comment by andreas-junghanns** [comment]
Why would the importer *set* a shift? The shift must be declated by the FMU and retreived by the import with getShift.
So I don´t think setShift is not needed, for any clock.

**Comment by KarlWernersson** [comment]
Is there a large difference compared to be able to set an interval?

It does complicates things as we might need a shift variability if we want to be fully flexible as one FMU might be able to adjust the interval but not the shift or the opposite.

If not, we should have an argument, at least "its complicated enough already" if there is no better.

If we don't allow to set a shift. Is shift only allowed for clocks with computed interval or can an FMU that the importer can set interval still have a shift? If not we need to explain that better.

**Comment by andreas-junghanns** [comment]
I think this late in the process we should not add flexibity - nobody asked for it, so it is likely not needed.
And if an FMU can adjust one of shift/interval, we should assume it can also the other.
But: The setInterval really is for information purpose only, as the actual clock activation comes from the importer.
As for the shift: we assumed that it is ok for the first tick to come as a "surprise", while the actual interval would likely make sense for some FMUs to do internal time estimations.

**Comment by KarlWernersson** [comment]
I am happy to avoid extra complexity if possible.

Does this mean that I should create setShiftDecimal and setShiftFraction functions for fixed and tunable periodic clocks ?
I think limiting setting shift to initialization would be good if we want to avoid extra complexity.

> But: The setInterval really is for information purpose only, as the actual clock activation comes from the importer.

Yes, but this information is very important if you work with clock fractions e.g 1/3 s. If working with doubles it is maybe easier to deduce.

> As for the shift: we assumed that it is ok for the first tick to come as a "surprise", while the actual interval would likely make sense for some FMUs to do internal time estimations.
 
Sorry I don't understand. Currently Shift is always known at instantiation through xml for **constant** **fixedComputed**(calculated) and **tunableComputed**(tunable) and can also be retrieved during initialization with getShift.

PS I am thinking on ME/CS clocks as I have little knowledge and experience of SE

**Comment by andreas-junghanns** [comment]
> Does this mean that I should create setShiftDecimal and setShiftFraction functions for fixed and tunable periodic clocks ? I think limiting setting shift to initialization would be good if we want to avoid extra complexity.

Yes, I think adding these functions makes sense and limiting to Initialization Mode also.
 
> Sorry I don't understand. Currently Shift is always known at instantiation through xml for **constant** **fixedComputed**(calculated) and **tunableComputed**(tunable) and can also be retrieved during initialization with getShift.

Yes, but for the other clock type(s) where setInterval is available. If you clean that up, I am sure that is ok as well.

@MBlesken and @PTaeuberDS : Please follow along here to make sure this is also working with your SE needs!!
